### PR TITLE
BUG-97053 V2 Connectors Regions test cases

### PR DESCRIPTION
**Pull request by aszegedi**
@afarsang Please review.

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    public ConnectorsTests(CloudProvider cp, TestParameter tp) {
+        this.cloudProvider = cp;
+        setTestParameter(tp);
+    }
+
+    @BeforeTest
+    @Parameters({ "provider" })
+    public void beforeTest(@Optional(OpenstackCloudProvider.OPENSTACK) String provider) {
+        LOGGER.info("before cluster test set provider: " + provider);
```
Comment (doktoric) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java]: LOGGER.info("before cluster test set provider:  {}", provider);

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+        try {
+            given(CloudbreakClient.isCreated());
+            given(Credential.request()
+                    .withName(cloudProvider.getCredentialName()));
+            when(Credential.delete());
+        } catch (ForbiddenException e) {
+            String exceptionMessage = e.getResponse().readEntity(String.class);
+            this.errorMessage = exceptionMessage.substring(exceptionMessage.lastIndexOf(":") + 1);
+            LOGGER.info("ForbiddenException message ::: " + this.errorMessage);
```
Comment (doktoric) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java]: LOGGER.info("ForbiddenException message {}", this.errorMessage);

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        given(CloudbreakClient.isCreated());
+        given(cloudProvider.aValidCredential(), cloudProvider.getPlatform() + " credential is created");
+        given(Region.request(), cloudProvider.getPlatform() + " region request");
+        when(Region.getPlatformRegions(), "Default region is requested to "
+                + cloudProvider.getPlatform() + " credential");
+        then(Region.assertThis(
+                (region, t) -> {
+                    LOGGER.info(cloudProvider.getPlatform() + " Default Region is ::: "
+                            + region.getRegionResponse().getDefaultRegion());
+                    Assert.assertTrue(region.getRegionResponse().getDefaultRegion().contains(cloudProvider.region()),
```
Comment (doktoric) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java]: can you please use string join ?

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            this.errorMessage = exceptionMessage.substring(exceptionMessage.lastIndexOf(":") + 1);
+            LOGGER.info("ForbiddenException message ::: " + this.errorMessage);
+        }
+    }
+
+    @Test(priority = 0, groups = { "regions" })
+    public void testDefaultRegionForCredential() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(cloudProvider.aValidCredential(), cloudProvider.getPlatform() + " credential is created");
+        given(Region.request(), cloudProvider.getPlatform() + " region request");
```
Comment (doktoric) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java]: can you please use string join ?

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    public void testListRegionsForCredential() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(cloudProvider.aValidCredential(), cloudProvider.getPlatform() + " credential is created");
+        given(Region.request(), cloudProvider.getPlatform() + " region request");
+        when(Region.getPlatformRegions(), "Regions are requested to "
+                + cloudProvider.getPlatform() + " credential");
+        then(Region.assertThis(
+                (region, t) -> {
+                    LOGGER.info(cloudProvider.getPlatform() + " number of Regions ::: "
+                            + Integer.toString(region.getRegionResponse().getRegions().size()));
```
Comment (doktoric) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ConnectorsTests.java]: please use logger pattern

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/Region.java [hunk 6]
```diff
@@ -1,62 +1,39 @@
...
-    }
-
-    public void withType(String type) {
-        this.type = type;
+    public void setPlatformResourceRequest(PlatformResourceRequestJson platformResourceRequest) {
+        this.platformResourceRequest = platformResourceRequest;
     }
 
-    public void setRegionRResponse(Collection<String> regionRResponse) {
-        this.regionRResponse = regionRResponse;
```
Comment (doktoric) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/Region.java]: regionRResponse or regionResponse

Comment (aszegedi) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/Region.java]: RegionRR and RegionAv are obsolete

**Pull request merged**